libraries/Microsoft.Bot.Builder/Inspection/InspectionActivityExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder
+{
+    public static class InspectionActivityExtensions
+    {
+        private static JsonSerializerSettings jsonSerializerSettings = new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore };
+
+        public static IEnumerable<Activity> Clone(this List<Activity> activities)
+        {
+            return activities.Select(activity => activity.Clone());
+        }
+
+        public static Activity Clone(this Activity activity)
+        {
+            return JsonConvert.DeserializeObject<Activity>(JsonConvert.SerializeObject(activity, jsonSerializerSettings));
+        }
```

Comment:
Given the json serialization settings above, it's not clear this would always work. Could you use the same settings found in the original activity deserializer? I *think* the right set to copy (or just consume) is this one:
[https://github.com/Microsoft/botbuilder-dotnet/blob/f0d355fcfe2e9f57c1019ead58cadd57fcfca253/libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/HttpHelper.cs#L18](https://github.com/Microsoft/botbuilder-dotnet/blob/f0d355fcfe2e9f57c1019ead58cadd57fcfca253/libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/HttpHelper.cs#L18)

That would apply the same rules to the Activity as when it was (typically) first pulled off the HTTP Stack.

Reply:
now everything is trace activities

Reply:
so no cloning